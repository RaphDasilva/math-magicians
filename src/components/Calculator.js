import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.btnClicked = (e) => {
      const clickedBtn = e.target.innerText;
      const Result = calculate(this.state, clickedBtn);
      this.setState(Result);
    };
  }

  render() {
    const { total, operation, next } = this.state;
    return (
      <div className="calculator">
        <div className="display">
          <span>
            {total}
            {operation}
            {next}
          </span>
        </div>
        <div className="keys">
          <button type="button" onClick={this.btnClicked}>AC</button>
          <button type="button" onClick={this.btnClicked}>+/-</button>
          <button type="button" onClick={this.btnClicked}>%</button>
          <button type="button" className="act" onClick={this.btnClicked}>รท</button>
        </div>
        <div className="keys">
          <button type="button" onClick={this.btnClicked}>7</button>
          <button type="button" onClick={this.btnClicked}>8</button>
          <button type="button" onClick={this.btnClicked}>9</button>
          <button type="button" className="act" onClick={this.btnClicked}>x</button>
        </div>
        <div className="keys">
          <button type="button" onClick={this.btnClicked}>6</button>
          <button type="button" onClick={this.btnClicked}>5</button>
          <button type="button" onClick={this.btnClicked}>4</button>
          <button type="button" className="act" onClick={this.btnClicked}>-</button>
        </div>
        <div className="keys">
          <button type="button" onClick={this.btnClicked}>3</button>
          <button type="button" onClick={this.btnClicked}>2</button>
          <button type="button" onClick={this.btnClicked}>1</button>
          <button type="button" className="act" onClick={this.btnClicked}>+</button>
        </div>
        <div className="keys">
          <button type="submit" className="zero" onClick={this.btnClicked}>0</button>
          <button type="button" className="dot" onClick={this.btnClicked}>.</button>
          <button type="button" className="dot act" onClick={this.btnClicked}>=</button>
        </div>
      </div>
    );
  }
}
export default Calculator;
