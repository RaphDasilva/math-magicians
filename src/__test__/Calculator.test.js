import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

test('expect home component to match snapshot', () => {
  const view = renderer.create(<Calculator />).toJSON();
  expect(view).toMatchSnapshot();
});

test('renders button tag', () => {
  render(<Calculator />);
  const linkElement = screen.getByRole('button', { name: 'AC' });
  expect(linkElement).toBeInTheDocument();
});

test('renders Ac in the UI', () => {
  render(<Calculator />);
  const linkElement = screen.getByText(/AC/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders Ac in the UI', () => {
  render(<Calculator />);
  const linkElement = screen.getByText('7');
  expect(linkElement).toBeInTheDocument();
});

test('should update the state when a button is clicked', () => {
  const { getByText } = render(<Calculator />);
  const button = getByText('7');
  fireEvent.click(button);
  expect(getByText('7')).toBeInTheDocument();
});
