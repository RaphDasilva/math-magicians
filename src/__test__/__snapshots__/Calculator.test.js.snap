// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect home component to match snapshot 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    <span>
      0
       
       
    </span>
  </div>
  <div
    className="keys"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="act"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="keys"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="act"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="keys"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="act"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="keys"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="act"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="keys"
  >
    <button
      className="zero"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="dot act"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
